### ISSUE TITLE
Failing notification send error

### ISSUE THREAD

**Issue by chadcrowell** [mediabeastnz/craft-commerce-abandoned-cart#23]
`2019-06-07 09:17:22 [-][1][-][error][craft\queue\QueueLogBehavior::afterError]  [4402] Send abandoned cart reminder (attempt: 1) - Error (time: 0.143s): No currency found with ISO code “”.` seems to be the main error. US dollar (USD) is set as the default currency, and that was default upon Commerce installation and hasn't been changed. 

```
2019-06-07 09:17:22 [-][1][-][info][mediabeastnz\abandonedcart\AbandonedCart::init] Abandoned Cart for Craft Commerce 2 plugin loaded
2019-06-07 09:17:22 [-][1][-][info][craft\queue\QueueLogBehavior::beforeExec]  [4402] Send abandoned cart reminder (attempt: 1) - Started
2019-06-07 09:17:22 [-][1][-][info][nystudio107\seomatic\services\{closure}] Meta container cache miss: actions/queue/retry/1
2019-06-07 09:17:22 [-][1][-][error][craft\queue\QueueLogBehavior::afterError]  [4402] Send abandoned cart reminder (attempt: 1) - Error (time: 0.143s): No currency found with ISO code “”.
2019-06-07 09:17:22 [-][1][-][error][Twig\Error\Error] Twig\Error\Error: No currency found with ISO code “”. in /home/swishvps/stemplecreek.com/templates/abandoned-cart-emails/first.twig:360
Stack trace:
#0 /home/swishvps/stemplecreek.com/vendor/craftcms/commerce/src/web/twig/Extension.php(60): craft\commerce\web\twig\Extension->_validatePaymentCurrency(NULL)
#1 /home/swishvps/stemplecreek.com/storage/runtime/compiled_templates/49/4931356cae82887c0b6a20ba4b750b2b66f39cc5349d2fbba0aebf4e7fca31d8.php(426): craft\commerce\web\twig\Extension->commerceCurrency('44.9900', NULL)
#2 /home/swishvps/stemplecreek.com/vendor/twig/twig/src/Template.php(382): __TwigTemplate_b8871b01d247fb25fde00998145a5764ecf8cb998208ce9a0909a798b3d3558e->doDisplay(Array, Array)
#3 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/web/twig/Template.php(52): Twig\Template->displayWithErrorHandling(Array, Array)
#4 /home/swishvps/stemplecreek.com/vendor/twig/twig/src/Template.php(359): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#5 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/web/twig/Template.php(34): Twig\Template->display(Array, Array)
#6 /home/swishvps/stemplecreek.com/vendor/twig/twig/src/Template.php(367): craft\web\twig\Template->display(Array)
#7 /home/swishvps/stemplecreek.com/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#8 /home/swishvps/stemplecreek.com/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#9 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/web/View.php(343): Twig\Environment->render('abandoned-cart-...', Array)
#10 /home/swishvps/stemplecreek.com/vendor/mediabeastnz/craft-commerce-abandoned-cart/src/services/Carts.php(263): craft\web\View->renderTemplate('abandoned-cart-...', Array)
#11 /home/swishvps/stemplecreek.com/vendor/mediabeastnz/craft-commerce-abandoned-cart/src/jobs/SendEmailReminder.php(62): mediabeastnz\abandonedcart\services\Carts->sendMail(Object(mediabeastnz\abandonedcart\models\AbandonedCart), 'You've left som...', 'vffarmgirl@gmai...', 'abandoned-cart-...')
#12 /home/swishvps/stemplecreek.com/vendor/yiisoft/yii2-queue/src/Queue.php(214): mediabeastnz\abandonedcart\jobs\SendEmailReminder->execute(Object(craft\queue\Queue))
#13 /home/swishvps/stemplecreek.com/vendor/yiisoft/yii2-queue/src/cli/Queue.php(147): yii\queue\Queue->handleMessage('4402', 'O:49:"mediabeas...', '300', 1)
#14 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/queue/Queue.php(96): yii\queue\cli\Queue->handleMessage('4402', 'O:49:"mediabeas...', '300', 1)
#15 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/controllers/QueueController.php(86): craft\queue\Queue->run()
#16 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/controllers/QueueController.php(106): craft\controllers\QueueController->actionRun()
#17 [internal function]: craft\controllers\QueueController->actionRetry()
#18 /home/swishvps/stemplecreek.com/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#19 /home/swishvps/stemplecreek.com/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#20 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/web/Controller.php(109): yii\base\Controller->runAction('retry', Array)
#21 /home/swishvps/stemplecreek.com/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('retry', Array)
#22 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/web/Application.php(297): yii\base\Module->runAction('queue/retry', Array)
#23 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/web/Application.php(565): craft\web\Application->runAction('queue/retry', Array)
#24 /home/swishvps/stemplecreek.com/vendor/craftcms/cms/src/web/Application.php(281): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#25 /home/swishvps/stemplecreek.com/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#26 /home/swishvps/stemplecreek.com/public_html/index.php(21): yii\base\Application->run()
#27 {main}
```

**Comment by mediabeastnz** [comment]
Hey @chadcrowell I'm not sure how you've tried to display the currency in your custom template but you should be able to use something like this https://docs.craftcms.com/commerce/v2/twig-filters.html#commercecurrency

Let me know if you can't get that to work

**Comment by chadcrowell** [comment]
Thanks. I copied your template directly, and just changed things like the company name and such. No code was changed.

```
{% for item in order.lineItems %}
                            <tr>
                                <td class="item-col item">
                                    <span style="color: #4d4d4d; font-weight:bold;"><a href="{{ item.purchasable.url }}{% if act %}&act={{ act }}{% endif %}">{{ item.description }}</a></span>
                                </td>
                                <td class="item-col quantity">
                                    {{ item.qty }} @ {% if item.onSale %}
                                            <strike>{{ item.price|commerceCurrency(cart.currency) }}</strike> {{ item.salePrice|commerceCurrency(cart.currency) }}
                                        {% else %}
                                            {{ item.price|commerceCurrency(cart.currency) }}
                                        {% endif %}
                                </td>
                                <td class="item-col">
                                    {{ item.subtotal|commerceCurrency(cart.currency) }}
                                </td>
                            </tr>
                            {% endfor %}
```

**Comment by mediabeastnz** [comment]
I see. In order to use those commerce "helper" functions you actually need access to an instance of commerce.

You can do that by adding something like this above that for loop:

```
{% if cart is not defined %}
    {% set cart = craft.commerce.carts.cart %}
{% endif %}
```

I've actually just released version 1.2.0 which now includes this as well a new test mode.
